---
title: "class_18v2"
author: "Robertha Amezola"
date: "May 30, 2019"
output: html_document
---

## R Markdown
```{r eval=FALSE}
BiocManager::install(c("GenomicDataCommons", "TCGAbiolinks", "maftools"))
```

```{r}
library(GenomicDataCommons)
library(TCGAbiolinks)
#library(maftools) will leave out for now
```

```{r}
GenomicDataCommons::status()
```

```{r}
cases_by_project <- cases() %>%
  facet("project.project_id") %>%
  aggregations()
head(cases_by_project)
```

```{r}
x <- cases_by_project$project.project_id

# Make a custom color vector for our plot
colvec <- rep("lightblue", nrow(x))
colvec[2] <- "red"

# Plot with 'log' for y axis and rotate labels with 'las'
#par(___)  
barplot(x$doc_count, names.arg=x$key, log="y", col=colvec, las=2)
```

```{r}
query <- GDCquery(project="TCGA-PAAD",
                  data.category="Transcriptome Profiling",
                  data.type="Gene Expression Quantification")

ans <- getResults(query)
```

```{r}
head(ans)
```

```{r}
View(ans)
```



#Hands on part 2

```{r}
library(bio3d)
seqs <- read.fasta("lecture18_sequences.fa")
seqs
```

#this is helps us find the sequence regions that have all 9-mer peptides 
#this will be done by aligning sequences



```{r}
ide <-conserv(seqs, method="identity")

ide
```
```{r}
miss.match.sites <-which(ide <1)
gap.sites<-gap.inspect(seqs)$t.inds
```

```{r}
tumor.sites = miss.match.sites[ !miss.match.sites %in% gap.sites]
```



```{r}
## Calculate positional identity scores
ide <- conserv(seqs$ali, method="identity")
mutant.sites <- which(ide < 1) 

## Exclude gap possitions from analysis
gaps <- gap.inspect(seqs)
mutant.sites <- mutant.sites[mutant.sites %in% gaps$f.inds]

mutant.sites
```

```{r}
## Make a "names" label for our output sequences (one per mutant)
mutant.names <- paste0(seqs$ali["P53_wt",mutant.sites],
                       mutant.sites,
                       seqs$ali["P53_mutant",mutant.sites])

mutant.names
```

```{r}
## Sequence positions surounding each mutant site
start.position <- mutant.sites - 8
end.position <-  mutant.sites + 8

# Blank matrix to store sub-sequences
store.seqs <- matrix("-", nrow=length(mutant.sites), ncol=17)
rownames(store.seqs) <- mutant.names

## Extract each sub-sequence
for(i in 1:length(mutant.sites)) {
  store.seqs[i,] <- seqs$ali["P53_mutant",start.position[i]:end.position[i]]
}

store.seqs
```

```{r}
## Output a FASTA file for further analysis
write.fasta(seqs=store.seqs, ids=mutant.names, file="subsequences.fa")
```

```{r}
library(bio3d)

## Read KRas oncogene sequence from UniProt
wt <- get.seq("P01116")

## Here we make four mutants namely: G12V, Q22N, T74S and A130V
mutant <- wt
mutant$ali[ c(12,22,74,130)] <- c("V", "N", "S", "V")

write.fasta( seqbind(wt, mutant), ids=c("wt","mutant"), file="kras-sequences.fa")
```


```{r}
start.position[1]: end.position[1]
```


```{r}
seqs$ali[2,start.position[1]: end.position[1]]
```






















