---
title: "class_05"
author: "Robertha Amezola"
date: "May 2, 2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

#Line plot
 
#weight
 
```{r}
read.csv("bimm143_05_rstats/weight_chart.txt")
```

```{r}
weight <- read.table("bimm143_05_rstats/weight_chart.txt", header=TRUE)
```
 
 
```{r}
plot(weight$Age, weight$Weight, typ="o", 
     pch=15, cex=1.5, lwd=2, ylim=c(2,10), 
     xlab="Age (months)", ylab="Weight (kg)", 
     main="Baby weight with age") 
```
 
 
 #2B barplot
 
```{r}
mouse.genome <- read.table("bimm143_05_rstats/feature_counts.txt", sep="\t", header=TRUE)

```
 
 
```{r}

barplot(mouse.genome$Count)
```
 
 
 
```{r}
par(mar=c(3.1, 11.1, 4.1, 2))
barplot(mouse.genome$Count, names.arg=mouse.genome$Feature, 
        horiz=TRUE, ylab="", 
        main="Number of features in the mouse GRCm38 genome", 
        las=1, xlim=c(0,80000))
```
 
 
 #2c Histograms
 
```{r}
x <- c(rnorm(10000),rnorm(10000)+4)
hist(x, breaks=80)
```
 
 # Using color in plots
 
 
```{r}
mf <- read.delim("bimm143_05_rstats/male_female_counts.txt")
```
 
 
```{r}
barplot(mf$Count, names.arg=mf$Sample, col=rainbow(nrow(mf)), 
        las=2, ylab="Counts")
```
 
 
 
```{r}
barplot(mf$Count, names.arg=mf$Sample, col=c("blue2","red2"), 
        las=2, ylab="Counts")


```

 
##3b coloring by value 

```{r}
genes <- read.delim("bimm143_05_rstats/up_down_expression.txt")
```


```{r}
table(genes$State)
```


```{r}
plot(genes$Condition1, genes$Condition2, col=genes$State, 
     xlab="Expression condition 1", ylab="Expression condition 2")
```

```{r}
palette(c("blue","gray","red"))
plot(genes$Condition1, genes$Condition2, col=genes$State, xlab="Expression condition 1", ylab="Expression condition 2")
```



#3C dynamic use fo color

```{r}
meth <- read.delim("bimm143_05_rstats/expression_methylation.txt")
```

```{r}
plot(meth$gene.meth, meth$expression)
```

# Plot changing the plot character ('pch') to a solid circle

```{r}
dcols <- densCols(meth$gene.meth, meth$expression)

plot(meth$gene.meth, meth$expression, col = dcols, pch = 20)
```


```{r}
inds <- meth$expression > 0
plot(meth$gene.meth[inds], meth$expression[inds])
```

```{r}
dcols <- densCols(meth$gene.meth[inds], meth$expression[inds])

plot(meth$gene.meth[inds], meth$expression[inds], col = dcols, pch = 20)
```


```{r}
dcols.custom <- densCols(meth$gene.meth[inds], meth$expression[inds],
                         colramp = colorRampPalette(c("blue2",
                                                      "green2",
                                                      "red2",
                                                      "yellow")) )

plot(meth$gene.meth[inds], meth$expression[inds], 
     col = dcols.custom, pch = 20)
```

#4 optiona; extension

```{r}
meth <- read.delim("bimm143_05_rstats/expression_methylation.txt")
```


```{r}
plot(meth$promoter.meth, meth$gene.meth, ylab="Gene Methylation", xlab="Promoter Methylation")
```

#source function


```{r}
source("bimm143_05_rstats/color_to_value_map.r")

mycols=map.colors(meth$expression, 
                  c(max(meth$expression), min(meth$expression)), 
                  colorRampPalette(c("blue","red"))(100))

plot(meth$promoter.meth, meth$gene.meth, 
     ylab="Gene Methylation", 
     xlab="Promoter Methylation", 
     col=mycols)
```



